src/bid/bid.rs

```diff
@@ -222,8 +222,45 @@ impl Bid {
             expiration_ts,
         })
     }
+
+    /// Perform a Poseidon sponge hash of all attributes of the Bid
+    pub fn hash(&self) -> BlsScalar {
+        let encrypted_data = self.encrypted_data.cipher().iter();
+
+        let data = [
+            self.nonce,
+            self.hashed_secret,
+            self.elegibility_ts,
+            self.expiration_ts,
+        ];
+
+        sponge_hash(
+            encrypted_data
+                .chain(data.iter())
+                .chain(self.stealth_address.R().to_hash_inputs().iter())
+                .chain(self.stealth_address.pk_r().to_hash_inputs().iter())
+                .chain(ExtendedPoint::from(self.c).to_hash_inputs().iter())
+                .map(|s| *s)
+                .collect::<Vec<BlsScalar>>()
+                .as_slice(),
+        )
+    }
```

Comment:
This function is not needed. Indeed we have the trait `From<Bid> for StorageScalar` which is doing that on the correct way. Since this way is not the correct one.